lib/ansible/modules/net_tools/nmcli.py

```diff
@@ -556,7 +560,7 @@ def __init__(self, module):
         self.type = module.params['type']
         self.ip4 = module.params['ip4']
         self.gw4 = module.params['gw4']
-        self.dns4 = ' '.join(module.params['dns4'])
+        self.dns4 = ' '.join(module.params['dns4']) if module.params.get('dns4') else None
```

Comment:
This is already in review - #30757

Reply:
I will remove it, this bug was hindering me so i copied this fix to my local.